src/controllers/hid/hiddevice.h

```diff
@@ -59,25 +64,46 @@ class DeviceInfo final {
         return m_serialNumberRaw.c_str();
     }
 
-    const QString& manufacturerString() const {
+    const QString& getVendorString() const {
         return m_manufacturerString;
     }
-    const QString& productString() const {
+    const QString& getProductString() const {
         return m_productString;
     }
-    const QString& serialNumber() const {
+    const QString& getSerialNumber() const {
         return m_serialNumber;
     }
 
-    bool isValid() const {
-        return !productString().isNull() && !serialNumber().isNull();
+    std::optional<uint8_t> getUsbInterfaceNumber() const {
+        if (m_usbInterfaceNumber == -1) {
+            return std::nullopt;
+        }
+        return m_usbInterfaceNumber;
+    }
+
+    const PhysicalTransportProtocol& getPhysicalTransportProtocol() const {
+        return m_physicalTransportProtocol;
+    }
+
+    uint16_t getUsagePage() const {
+        return usage_page;
+    }
+
+    uint16_t getUsage() const {
+        return usage;
+    }
+
+    QString getUsagePageDescription() const {
+        return mixxx::hid::HidUsageTables::getUsagePageDescription(usage_page);
+    }
+
+    QString getUsageDescription() const {
+        return mixxx::hid::HidUsageTables::getUsageDescription(usage_page, usage);
     }
 
-    QString formatVID() const;
-    QString formatPID() const;
-    QString formatReleaseNumber() const;
-    QString formatInterface() const;
-    QString formatUsage() const;
+    bool isValid() const {
+        return !getProductString().isNull() && !getSerialNumber().isNull();
+    }
```

Comment:
are you sure all hid devices have a productstring and serial number they report? Why do exactly these attributes make the device "valid"? What does valid mean in this context? Usually objects that are not valid result in UB if you call most member functions on them, but thats not the case here, afaict.

Reply:
I didn't touch this line in this PR - and do not plan to.

But I can tell you, that only USB HID devices can store a product string. But the numeric VID & PID exist for all HID devices. This is because the HID descriptors are not specified with fields for strings. They just contain the numerical index of a string in the USB String Descriptor. And non-USB devices simply have no USB String Descriptor to reference.
But it seems the fields are always filled by the OS, in case of my Lenovo Thinkpad with an I2C Touchpad from Elan, Windows reports "Microsoft" "HIDI2C Device" with S/N 9999.

Reply:
whoops yeah sorry about the complaint. The diff split weirdly so I didn't notice that this was already here previously. Then disregard the comment.